C-extension compiler warnings

When you don't use tox, you see this in the build log:

```
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/cdrizzleapi.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/cdrizzleapi.o
  /home/runner/work/drizzle/drizzle/src/cdrizzleapi.c: In function ‘PyInit_cdrizzle’:
  /home/runner/work/drizzle/drizzle/src/cdrizzleapi.c:610:12: warning: returning ‘PyObject *’ {aka ‘struct _object *’} from a function with incompatible return type ‘PyObject **’ {aka ‘struct _object **’} [-Wincompatible-pointer-types]
    610 |     return m;
        |            ^
  gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/cdrizzleblot.o
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c: In function ‘interpolate_poly3’:
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:[120](https://github.com/spacetelescope/drizzle/actions/runs/7014084246/job/19081247879?pr=128#step:4:121):47: warning: ‘cd21[3]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    120 |       ztemp[j] = sx * (coeff[index+1] + sx2m1 * cd21[j]) +
        |                                         ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:18: note: ‘cd21[3]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |                  ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:120:47: warning: ‘cd21[2]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    120 |       ztemp[j] = sx * (coeff[index+1] + sx2m1 * cd21[j]) +
        |                                         ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:18: note: ‘cd21[2]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |                  ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:120:47: warning: ‘cd21[1]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    120 |       ztemp[j] = sx * (coeff[index+1] + sx2m1 * cd21[j]) +
        |                                         ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:18: note: ‘cd21[1]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |                  ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:120:47: warning: ‘cd21[0]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    120 |       ztemp[j] = sx * (coeff[index+1] + sx2m1 * cd21[j]) +
        |                                         ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:18: note: ‘cd21[0]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |                  ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:[121](https://github.com/spacetelescope/drizzle/actions/runs/7014084246/job/19081247879?pr=128#step:4:122):45: warning: ‘cd20[3]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    121 |                  tx * (coeff[index] + tx2m1 * cd20[j]);
        |                                       ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:9: note: ‘cd20[3]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |         ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:121:45: warning: ‘cd20[2]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    121 |                  tx * (coeff[index] + tx2m1 * cd20[j]);
        |                                       ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:9: note: ‘cd20[2]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |         ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:121:45: warning: ‘cd20[1]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    121 |                  tx * (coeff[index] + tx2m1 * cd20[j]);
        |                                       ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:9: note: ‘cd20[1]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |         ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:121:45: warning: ‘cd20[0]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    121 |                  tx * (coeff[index] + tx2m1 * cd20[j]);
        |                                       ~~~~~~^~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:72:9: note: ‘cd20[0]’ was declared here
     72 |   float cd20[4], cd21[4], ztemp[4];
        |         ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c: In function ‘interpolate_sinc’:
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:673:19: warning: array subscript -1 is below array bounds of ‘float[15]’ [-Warray-bounds]
    673 |         px = taper[j - 1] / ax;
        |              ~~~~~^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:596:9: note: while referencing ‘taper’
    596 |   float taper[INTERPOLATE_SINC_NCONV];
        |         ^~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:681:19: warning: array subscript -1 is below array bounds of ‘float[15]’ [-Warray-bounds]
    681 |         py = taper[j - 1] / ay;
        |              ~~~~~^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:596:9: note: while referencing ‘taper’
    596 |   float taper[INTERPOLATE_SINC_NCONV];
        |         ^~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:685:9: warning: array subscript -1 is below array bounds of ‘float[15]’ [-Warray-bounds]
    685 |       ac[j - 1] = px;
        |       ~~^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:597:9: note: while referencing ‘ac’
    597 |   float ac[INTERPOLATE_SINC_NCONV], ar[INTERPOLATE_SINC_NCONV];
        |         ^~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:686:9: warning: array subscript -1 is below array bounds of ‘float[15]’ [-Warray-bounds]
    686 |       ar[j - 1] = py;
        |       ~~^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzleblot.c:597:37: note: while referencing ‘ar’
    597 |   float ac[INTERPOLATE_SINC_NCONV], ar[INTERPOLATE_SINC_NCONV];
        |                                     ^~
  gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/cdrizzlebox.o
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c: In function ‘do_kernel_point’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c:254:15: warning: unused variable ‘ybounds’ [-Wunused-variable]
    254 |     integer_t ybounds[2], osize[2];
        |               ^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c: In function ‘do_kernel_tophat’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c:343:15: warning: unused variable ‘ybounds’ [-Wunused-variable]
    343 |     integer_t ybounds[2], osize[2];
        |               ^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c: In function ‘do_kernel_gaussian’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c:461:15: warning: unused variable ‘ybounds’ [-Wunused-variable]
    461 |     integer_t ybounds[2], osize[2];
        |               ^~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c: In function ‘do_kernel_lanczos’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c:589:15: warning: unused variable ‘ybounds’ [-Wunused-variable]
    589 |     integer_t ybounds[2], osize[2];
        |               ^~~~~~~
  At top level:
  /home/runner/work/drizzle/drizzle/src/cdrizzlebox.c:15:13: warning: ‘buf’ defined but not used [-Wunused-variable]
     15 | static char buf[1024];
        |             ^~~
  gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/cdrizzlemap.o
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c: In function ‘interpolate_point’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:108:27: warning: unused variable ‘idim’ [-Wunused-variable]
    108 |     int ipix, jpix, npix, idim;
        |                           ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:108:21: warning: unused variable ‘npix’ [-Wunused-variable]
    108 |     int ipix, jpix, npix, idim;
        |                     ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:108:15: warning: unused variable ‘jpix’ [-Wunused-variable]
    108 |     int ipix, jpix, npix, idim;
        |               ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:108:9: warning: unused variable ‘ipix’ [-Wunused-variable]
    108 |     int ipix, jpix, npix, idim;
        |         ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c: In function ‘map_point’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:201:15: warning: variable ‘status’ set but not used [-Wunused-but-set-variable]
    201 |     int i, j, status;
        |               ^~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c: In function ‘clip_polygon_to_window’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:501:16: warning: passing argument 1 of ‘orient_ccw’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
    501 |     orient_ccw(p);
        |                ^
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:438:28: note: expected ‘struct polygon *’ but argument is of type ‘const struct polygon *’
    438 | orient_ccw(struct polygon *p) {
        |            ~~~~~~~~~~~~~~~~^
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:502:16: warning: passing argument 1 of ‘orient_ccw’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
    502 |     orient_ccw(wnd);
        |                ^~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:438:28: note: expected ‘struct polygon *’ but argument is of type ‘const struct polygon *’
    438 | orient_ccw(struct polygon *p) {
        |            ~~~~~~~~~~~~~~~~^
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:495:21: warning: unused variable ‘signed_area’ [-Wunused-variable]
    495 |     double t, u, d, signed_area, app_, aww_;
        |                     ^~~~~~~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:495:15: warning: unused variable ‘u’ [-Wunused-variable]
    495 |     double t, u, d, signed_area, app_, aww_;
        |               ^
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:495:12: warning: unused variable ‘t’ [-Wunused-variable]
    495 |     double t, u, d, signed_area, app_, aww_;
        |            ^
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c: In function ‘init_image_scanner’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:911:9: warning: unused variable ‘ipoint’ [-Wunused-variable]
    911 |     int ipoint;
        |         ^~~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:910:25: warning: unused variable ‘osize’ [-Wunused-variable]
    910 |     integer_t isize[2], osize[2];
        |                         ^~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:910:15: warning: unused variable ‘isize’ [-Wunused-variable]
    910 |     integer_t isize[2], osize[2];
        |               ^~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:909:21: warning: unused variable ‘xyout’ [-Wunused-variable]
    909 |     double xyin[2], xyout[2];
        |                     ^~~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:909:12: warning: unused variable ‘xyin’ [-Wunused-variable]
    909 |     double xyin[2], xyout[2];
        |            ^~~~
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c: In function ‘map_point’:
  /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:216:1: warning: control reaches end of non-void function [-Wreturn-type]
    216 | }
        | ^
  In file included from /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:21,
                   from /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/arrayobject.h:5,
                   from /home/runner/work/drizzle/drizzle/src/cdrizzlemap.c:10:
  At top level:
  /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/__multiarray_api.h:[145](https://github.com/spacetelescope/drizzle/actions/runs/7014084246/job/19081247879?pr=128#step:4:146)8:1: warning: ‘_import_array’ defined but not used [-Wunused-function]
   1458 | _import_array(void)
        | ^~~~~~~~~~~~~
  gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/cdrizzleutil.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/cdrizzleutil.o
  /home/runner/work/drizzle/drizzle/src/cdrizzleutil.c: In function ‘create_lanczos_lut’:
  /home/runner/work/drizzle/drizzle/src/cdrizzleutil.c:277:17: warning: comparison of integer expressions of different signedness: ‘integer_t’ {aka ‘int’} and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
    277 |   for (i = 1; i < npix; ++i) {
        |                 ^
  In file included from /usr/include/string.h:535,
                   from /opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12/Python.h:26,
                   from /home/runner/work/drizzle/drizzle/src/cdrizzleutil.h:5,
                   from /home/runner/work/drizzle/drizzle/src/cdrizzlemap.h:5,
                   from /home/runner/work/drizzle/drizzle/src/cdrizzleutil.c:3:
  In function ‘strncpy’,
      inlined from ‘driz_error_set_message’ at /home/runner/work/drizzle/drizzle/src/cdrizzleutil.c:39:3:
  /usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10: warning: ‘__builtin_strncpy’ specified bound 512 equals destination size [-Wstringop-truncation]
     95 |   return __builtin___strncpy_chk (__dest, __src, __len,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     96 |                                   __glibc_objsize (__dest));
        |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include -I/home/runner/work/drizzle/drizzle/src -I/opt/hostedtoolcache/Python/3.12.0/x64/include/python3.12 -c /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c -o /tmp/tmp3_q9wi6l.build-temp/home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.o
  /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c: In function ‘utest_cdrizzle’:
  /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:618:23: warning: unused variable ‘ybounds’ [-Wunused-variable]
    618 |             integer_t ybounds[2];       /* start of in-bounds */
        |                       ^~~~~~~
  /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:642:23: warning: unused variable ‘ybounds’ [-Wunused-variable]
    642 |             integer_t ybounds[2];       /* start of in-bounds */
        |                       ^~~~~~~
  In file included from /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/ndarrayobject.h:21,
                   from /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/arrayobject.h:5,
                   from /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:9:
  At top level:
  /opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/numpy/_core/include/numpy/__multiarray_api.h:1458:1: warning: ‘_import_array’ defined but not used [-Wunused-function]
   1458 | _import_array(void)
        | ^~~~~~~~~~~~~
  In file included from /usr/include/string.h:535,
                   from /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:4:
  In function ‘strncpy’,
      inlined from ‘fctstr_safe_cpy’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:267:5,
      inlined from ‘fct_test_new’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:974:5:
  /usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10: warning: ‘__builtin_strncpy’ specified bound [256](https://github.com/spacetelescope/drizzle/actions/runs/7014084246/job/19081247879?pr=128#step:4:257) equals destination size [-Wstringop-truncation]
     95 |   return __builtin___strncpy_chk (__dest, __src, __len,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     96 |                                   __glibc_objsize (__dest));
        |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  In file included from /home/runner/work/drizzle/drizzle/src/tests/pandokia_fct.h:25,
                   from /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:15:
  In function ‘fct_nlist__clear’,
      inlined from ‘fct_nlist__final’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:751:5,
      inlined from ‘fct_test__del’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:956:5,
      inlined from ‘fct_test__del’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:950:1,
      inlined from ‘fct_test_new’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:991:9:
  /home/runner/work/drizzle/drizzle/src/tests/fct.h:736:41: warning: ‘((fct_nlist_t *)test)[1].used_itm_num’ may be used uninitialized [-Wmaybe-uninitialized]
    736 |         for ( itm_i__=0; itm_i__ != list->used_itm_num; ++itm_i__ )
        |                                     ~~~~^~~~~~~~~~~~~~
  In file included from /usr/include/string.h:535,
                   from /home/runner/work/drizzle/drizzle/src/tests/utest_cdrizzle.c:4:
  In function ‘strncpy’,
      inlined from ‘fctstr_safe_cpy’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:[267](https://github.com/spacetelescope/drizzle/actions/runs/7014084246/job/19081247879?pr=128#step:4:268):5,
      inlined from ‘fctchk_new’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:886:5,
      inlined from ‘_fct_xchk_fn_varg’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:3581:11,
      inlined from ‘fct_xchk_fn’ at /home/runner/work/drizzle/drizzle/src/tests/fct.h:3624:9:
  /usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10: warning: ‘__builtin_strncpy’ specified bound 256 equals destination size [-Wstringop-truncation]
     95 |   return __builtin___strncpy_chk (__dest, __src, __len,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     96 |                                   __glibc_objsize (__dest));
        |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/setup.py
+++ b/setup.py
@@ -25,13 +25,20 @@ def get_extensions():
     }
     cfg['include_dirs'].append(numpy.get_include())
     cfg['include_dirs'].append(srcdir)
-    if sys.platform != 'win32':
-        cfg['libraries'].append('m')
+
     if sys.platform == 'win32':
         cfg['define_macros'].append(('WIN32', None))
         cfg['define_macros'].append(('__STDC__', 1))
         cfg['define_macros'].append(('_CRT_SECURE_NO_WARNINGS', None))
-
+    else:
+        cfg['libraries'].append('m')
+        cfg['extra_compile_args'] = [
+            '-Wall',
+            '-Wextra',
+            '-Wpedantic',
+            '-Wno-unused-parameter',
+            '-Wincompatible-pointer-types'
+        ]
     # importing these extension modules is tested in `.github/workflows/build.yml`;
     # when adding new modules here, make sure to add them to the `test_command` entry there
     return [Extension(str('drizzle.cdrizzle'), sources, **cfg)]

--- a/src/cdrizzleapi.c
+++ b/src/cdrizzleapi.c
@@ -139,7 +139,7 @@ tdriz(PyObject *obj UNUSED_PARAM, PyObject *args, PyObject *keywords) {
             driz_error_set_message(&error, "Invalid context array");
             goto _exit;
         }
-    };
+    }
 
     /* Convert the fill value string */
 
@@ -607,10 +607,13 @@ clip_polygon_wrap(PyObject *self, PyObject *args) {
 /** ---------------------------------------------------------------------------
  * Table of functions callable from python
  */
+#if defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#elif defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wcast-function-type-mismatch"
+#endif
 static struct PyMethodDef cdrizzle_methods[] = {
     {"tdriz", (PyCFunction)tdriz, METH_VARARGS | METH_KEYWORDS,
      "tdriz(image, weights, pixmap, output, counts, context, uniqid, xmin, "
@@ -626,8 +629,11 @@ static struct PyMethodDef cdrizzle_methods[] = {
     {"clip_polygon", clip_polygon_wrap, METH_VARARGS, "clip_polygon(p, q)"},
     {NULL, NULL} /* sentinel */
 };
-#pragma clang diagnostic pop
+#if defined(__GNUC__)
 #pragma GCC diagnostic pop
+#elif defined(__clang__)
+#pragma clang diagnostic pop
+#endif
 
 /** ---------------------------------------------------------------------------
  */

--- a/src/cdrizzleapi.c
+++ b/src/cdrizzleapi.c
@@ -607,7 +607,10 @@ clip_polygon_wrap(PyObject *self, PyObject *args) {
 /** ---------------------------------------------------------------------------
  * Table of functions callable from python
  */
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-function-type-mismatch"
 static struct PyMethodDef cdrizzle_methods[] = {
     {"tdriz", (PyCFunction)tdriz, METH_VARARGS | METH_KEYWORDS,
      "tdriz(image, weights, pixmap, output, counts, context, uniqid, xmin, "
@@ -623,6 +626,8 @@ static struct PyMethodDef cdrizzle_methods[] = {
     {"clip_polygon", clip_polygon_wrap, METH_VARARGS, "clip_polygon(p, q)"},
     {NULL, NULL} /* sentinel */
 };
+#pragma clang diagnostic pop
+#pragma GCC diagnostic pop
 
 /** ---------------------------------------------------------------------------
  */

--- a/src/cdrizzlemap.c
+++ b/src/cdrizzlemap.c
@@ -592,7 +592,7 @@ init_edge(struct edge *e, struct vertex v1, struct vertex v2, int position) {
     e->m = (v2.x - v1.x) / (v2.y - v1.y);
     e->b = (v1.x * v2.y - v1.y * v2.x) / (v2.y - v1.y);
     e->c = e->b - copysign(0.5 + 0.5 * fabs(e->m), (double)position);
-};
+}
 
 /**
  * Set-up scanner structure for a polygon.
@@ -763,7 +763,7 @@ get_scanline_limits(struct scanner *s, int y, int *x1, int *x2) {
             return 1;
         }
         ++s->left;
-    };
+    }
 
     while (pyb > s->right->v2.y) {
         if (s->right == er_max) {
@@ -772,7 +772,7 @@ get_scanline_limits(struct scanner *s, int y, int *x1, int *x2) {
             return 1;
         }
         ++s->right;
-    };
+    }
 
     xlb = s->left->m * y + s->left->c - MAX_INV_ERR;
     xrb = s->right->m * y + s->right->c + MAX_INV_ERR;
@@ -786,7 +786,7 @@ get_scanline_limits(struct scanner *s, int y, int *x1, int *x2) {
         }
         ++s->left;
         edge_ymax = s->left->v2.y + 0.5 + MAX_INV_ERR;
-    };
+    }
 
     edge_ymax = s->right->v2.y + 0.5 + MAX_INV_ERR;
     while (pyt > edge_ymax) {
@@ -797,7 +797,7 @@ get_scanline_limits(struct scanner *s, int y, int *x1, int *x2) {
         }
         ++s->right;
         edge_ymax = s->right->v2.y + 0.5 + MAX_INV_ERR;
-    };
+    }
 
     xlt = s->left->m * y + s->left->c - MAX_INV_ERR;
     xrt = s->right->m * y + s->right->c + MAX_INV_ERR;

--- a/src/cdrizzleutil.c
+++ b/src/cdrizzleutil.c
@@ -237,7 +237,7 @@ bool2str(bool_t value) {
 void
 create_lanczos_lut(const int kernel_order, const size_t npix, const float del,
                    float *lanczos_lut) {
-    integer_t i;
+    size_t i;
     const float forder = (float)kernel_order;
     float poff;
 

--- a/src/tests/pandokia_fct.h
+++ b/src/tests/pandokia_fct.h
@@ -166,10 +166,10 @@ pandokia_skip (fct_logger_i * li, fct_logger_evt_t const *e)
 
 /*
  * pandokia_logger_object serves two purposes:
- * 
+ *
  * - a flag that indicates we are running in pandokia
  * - a shortcut to the actual pandokia data
- * 
+ *
  * Otherwise, you would have to walk some fairly complicated data
  * structures inside fctx to get to this single instance.
  */
@@ -182,7 +182,7 @@ struct _pandokia_logger *pandokia_logger_object = NULL;
 fct_logger_i *
 pandokia_logger (void)
 {
-	struct _pandokia_logger *l = 
+	struct _pandokia_logger *l =
 		(struct _pandokia_logger *) calloc (1, sizeof ( struct _pandokia_logger));
 	if (l == NULL)
 		return NULL;
@@ -223,7 +223,7 @@ pandokia_logger (void)
 	l->pdk_basename = strdup(l->pdk_file);	/* leaked */
 	assert( l->pdk_basename );
 	{ char *s = strrchr(l->pdk_basename,'.'); if (s) *s=0; }
-	
+
 	return (fct_logger_i *) l;
 }
 
@@ -240,7 +240,7 @@ pandokia_logger (void)
  */
 
 static void
-pandokia_intercept_logger ()
+pandokia_intercept_logger(void)
 {
 	char *s;
 	/*
@@ -380,7 +380,7 @@ void pandokia_okfile_real( fctkern_t *fctkern_ptr__, char *filename)
 
 		/* */
 		free( okfile_name );
-		
+
 		}
 
 	/*

--- a/src/tests/utest_cdrizzle.c
+++ b/src/tests/utest_cdrizzle.c
@@ -241,7 +241,7 @@ print_pixmap(char *title, struct driz_param_t *p, int lo, int hi) {
 }
 
 struct driz_param_t *
-setup_parameters() {
+setup_parameters(void) {
     struct driz_error_t *error;
 
     /* Initialize the parameter struct with vanilla defaults */
@@ -1132,4 +1132,6 @@ FCT_BGN_FN(utest_cdrizzle) {
     }
     FCT_FIXTURE_SUITE_END();
 }
+{
 FCT_END_FN();
+}

